spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/partitioner/ReplicaPartitioner.scala

```diff
@@ -38,14 +62,20 @@ class ReplicaPartitioner(partitionsPerReplicaSet: Int, val connector: CassandraC
    */
   override def getPartition(key: Any): Int = {
     key match {
-      case key: Set[_] if key.size > 0 && key.forall(_.isInstanceOf[InetAddress]) =>
+      case key: T =>
```

Comment:
I pass in the classtag? Isn't that enough. I don't see a warning in the compiler?